nailgun/entity_mixins.py

```diff
@@ -204,6 +205,77 @@ def _get_server_config():
     return config.ServerConfig.get()
 
 
+def _get_entity_id(field_name, attrs):
```

Comment:
This method's docstring is worth reading. It provides examples of the different ways in which the server might hand back a response. This method is at the heart of this PR.